common/src/main/java/io/pravega/common/health/processor/impl/SocketStreamHealthRequestProcessorImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.health.processor.impl;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.string.StringDecoder;
+import io.netty.handler.codec.string.StringEncoder;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+import java.io.DataOutputStream;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A HealthRequestProcessor driven from a socket stream.
+ */
+@Slf4j
+public class SocketStreamHealthRequestProcessorImpl extends HealthRequestProcessorImpl {
+    private final int port;
+    private EventLoopGroup bossGroup;
+    private EventLoopGroup workerGroup;
+    private boolean nio = false;
+    private Channel serverChannel;
+    private String host;
+
+    public SocketStreamHealthRequestProcessorImpl(String host, int port) {
+        super();
+        this.host = host;
+        this.port = port;
+    }
+
+    public void startListening() {
+        try {
+            bossGroup = new EpollEventLoopGroup(1);
+            workerGroup = new EpollEventLoopGroup();
+        } catch (ExceptionInInitializerError | NoClassDefFoundError e) {
```

Comment:
This code will break with the latest netty as they change the exception that is thrown. Perhaps we should make a static method somewhere to do this so as not to duplicate code.